From 4cfab865ccfc4d6f1d51b97f5edcf156ec142c28 Mon Sep 17 00:00:00 2001
From: Chenhui Zhao <chenhui.zhao@freescale.com>
Date: Wed, 5 Nov 2014 16:25:20 +0800
Subject: [PATCH 031/129] pm: ls1021: fix deep sleep issue when compiling
 kernel with THUMB2 instuction

* set the bit 0 of the target address of long jump to 1 for THUMB mode
* compile the resume entry code in arm instruction set

Signed-off-by: Chenhui Zhao <chenhui.zhao@freescale.com>
Change-Id: I01a96158ac39e14dcaebc7305b03eb277712011b
Reviewed-on: http://git.am.freescale.net:8181/23209
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Zhengxiong Jin <Jason.Jin@freescale.com>
[Xulin: Original patch taken from
Freescale-Linux-SDK-for-LS1021A-IOT-Rev2-v0.4-SOURCE-20150907-yocto.iso]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 arch/arm/mach-imx/sleep-ls1.S |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-imx/sleep-ls1.S b/arch/arm/mach-imx/sleep-ls1.S
index ae0a8af..b8d7f1a 100644
--- a/arch/arm/mach-imx/sleep-ls1.S
+++ b/arch/arm/mach-imx/sleep-ls1.S
@@ -59,6 +59,7 @@ ENTRY(ls1_do_deepsleep)
 	isb
 
 	/* jump to sram code using physical address */
+ THUMB( orr r0, r0, #1 )
 	bx	r0
 ENDPROC(ls1_do_deepsleep)
 	.popsection
@@ -123,6 +124,7 @@ ENTRY(ls1_sram_code_size)
 	.word	. - ls1_start_fsm
 
 /* the bootloader will jump to here after wakeup from deep sleep  */
+	.arm
 	.align L1_CACHE_SHIFT
 ENTRY(ls1_deepsleep_resume)
  THUMB(	adr	r6, BSYM(1f)	)
-- 
1.7.5.4

